gpt4all-chat/logger.h

```diff
@@ -0,0 +1,18 @@
+#ifndef LOGGER_H
+#define LOGGER_H
+#include <QFile>
+
+class Logger
+{
+    QFile m_file;
+
+    static void messageHandler(QtMsgType type, const QMessageLogContext &context, const QString &msg);
+
+public:
+    static Logger *globalInstance();
+
+    explicit Logger();
+    friend class MyLogger;
```

Comment:
You don't need this if you're not going to use the Q_GLOBAL_STATIC?? why did you remove that? I like this because it is guaranteed thread safe on all platforms... you don't need to inherit QObject to use the previous version of singleton that you had...

Reply:
Hmm, my experience with singletons is quite limited, because I always wanted to avoid them. Leaving in `MyLogger` was an oversight that is fixed now, thanks!